src/NServiceBus.Core/Routing/IUnicastRouter.cs

```diff
@@ -2,10 +2,11 @@ namespace NServiceBus.Routing
 {
     using System;
     using System.Collections.Generic;
+    using System.Threading.Tasks;
     using NServiceBus.Extensibility;
 
     interface IUnicastRouter
     {
-        IEnumerable<UnicastRoutingStrategy> Route(Type messageType, DistributionStrategy distributionStrategy, ContextBag contextBag);
+        Task<IEnumerable<UnicastRoutingStrategy>> Route(Type messageType, DistributionStrategy distributionStrategy, ContextBag contextBag);
```

Comment:
Does it need to be `IEnumerable`? List, ReadonlyCollection whatever would be better